cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ScheduleSpecificationCodec.java

```diff
@@ -47,7 +86,28 @@ public ScheduleSpecification deserialize(JsonElement json, Type typeOfT, JsonDes
     throws JsonParseException {
     JsonObject jsonObj = json.getAsJsonObject();
 
-    Schedule schedule = context.deserialize(jsonObj.get("schedule"), Schedule.class);
+    JsonElement scheduleTypeJson = jsonObj.get("scheduleType");
+    ScheduleType scheduleType;
+    if (scheduleTypeJson == null) {
+      // For backwards compatibility with spec persisted with older versions than 2.8, we need these lines
+      scheduleType = ScheduleType.ORIGINAL_TIME;
+    } else {
+      scheduleType = context.deserialize(jsonObj.get("scheduleType"), ScheduleType.class);
+    }
+
+    Schedule schedule = null;
+    switch (scheduleType) {
+      case ORIGINAL_TIME:
+        schedule = context.deserialize(jsonObj.get("schedule"), Schedule.class);
+        break;
```

Comment:
Ok i think I see what you mean now, that because the configurer only creates ``TimeSchedule`` then there will be no ``Schedule`` here. But if an app has been created prior to 2.8.0, its serialized spec will contain Schedule objects, and so the deserialization still needs to handle the ``ORIGINAL_TIME``.